### PR TITLE
[risk=no][RW-4896] New User Profile e2e tests

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test:local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: e2e/tests/profile/profile-page.spec.ts [hunk 1]
```diff
@@ -1,26 +1,123 @@
...
+    firstName = await profilePage.getFirstNameInput();
+    lastName = await profilePage.getLastNameInput();
+    url = await profilePage.getProfessionalUrlInput();
+    researchBackground = await profilePage.getResearchBackgroundTextarea();
+    address1 = await profilePage.getAddress1Input();
+    address2 = await profilePage.getAddress2Input();
+    city = await profilePage.getCityInput();
+    state = await profilePage.getStateInput();
+    zip = await profilePage.getZipCodeInput();
+    country = await profilePage.getCountryInput();
```
Comment (aweng98) [inline e2e/tests/profile/profile-page.spec.ts:92]: refinding all elements are not necessary.

File: e2e/tests/profile/profile-page.spec.ts [hunk 2]
```diff
@@ -1,26 +1,123 @@
-import HomePage from 'app/page/home-page';
 import ProfilePage from 'app/page/profile-page';
 import {signIn} from 'utils/test-utils';
 import navigation, {NavLink} from 'app/component/navigation';
+import {makeString, makeUrl} from 'utils/str-utils';
+import Button from '../../app/element/button';
```
Comment (aweng98) [inline e2e/tests/profile/profile-page.spec.ts:5]: no relative path in import. `../../` can be removed.

File: e2e/app/page/profile-page.ts [hunk 3]
```diff
@@ -1,25 +1,32 @@
-import {Page} from 'puppeteer';
+import {Page, WaitForSelectorOptions} from 'puppeteer';
 import Textbox from 'app/element/textbox';
 import AuthenticatedPage from 'app/page/authenticated-page';
 import {waitWhileLoading} from 'utils/test-utils';
 import {waitForDocumentTitle, waitForUrl} from 'utils/waits-utils';
+import Button from '../element/button';
+import Textarea from '../element/textarea';
+import BaseElement from '../element/base-element';
```
Comment (aweng98) [inline e2e/app/page/profile-page.ts:8]: type app instead .. in import statements. do not use relative path.

File: e2e/app/page/profile-page.ts [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
-  FirstName: 'First Name',
-  LastName: 'Last Name',
-  ContactEmail: 'Contact Email',
-  CurrentPosition: 'Your Current Position',
-  Organization: 'Your Organization',
-  CurrentResearchWork: 'Current Research Work',
-  AboutYou: 'About You',
-  Institution: 'Institution',
-  Role: 'Role',
-  DiscardChanges: 'Discard Changes',
```
Comment (aweng98) [inline e2e/app/page/profile-page.ts:19]: One disadvantage of not using text labels to find elements is no checking of text labels.

File: e2e/app/page/profile-page.ts [hunk 5]
```diff
@@ -1,25 +1,32 @@
...
+  ResearchBackground: 'Your research background, experience and research interests',
   SaveProfile: 'Save Profile',
 };
 
+
+export const DataTestIdAlias = {
```
Comment (aweng98) [inline e2e/app/page/profile-page.ts:18]: `export` is not needed if `DataTestIdAlias` isn't going to be imported by any outside file.

File: e2e/app/page/profile-page.ts [hunk 6]
```diff
@@ -40,12 +47,64 @@ export default class ProfilePage extends AuthenticatedPage {
...
+    const handle = await this.page.waitForXPath(selector, options);
+    return BaseElement.asBaseElement(this.page, handle);
+  }
+
+  async expectResearchPurposeErrorMissing(): Promise<void> {
+    const text = 'Current Research can\'t be blank';
+    await this.getDivWithText(text, {hidden: true});
+  }
+
+  async expectResearchPurposeErrorPresent(): Promise<void> {
```
Comment (aweng98) [inline e2e/app/page/profile-page.ts:106]: `await waitForText(this.page, text)` can be used to find texts on page.

File: e2e/tests/profile/profile-page.spec.ts [hunk 7]
```diff
@@ -1,26 +1,123 @@
...
+    // Type in Research Background textarea
+    let researchBackground = await profilePage.getResearchBackgroundTextarea();
+    await researchBackground.paste(testText);
+
+    // profile update should enable Save button
+    const saveButton = await waitForSaveButton(true);
+    await saveButton.click();
+    await profilePage.waitForLoad();
+
+    researchBackground = await profilePage.getResearchBackgroundTextarea();
```
Comment (aweng98) [inline e2e/tests/profile/profile-page.spec.ts:40]: I believe refinding is not necessary.

File: e2e/app/page/profile-page.ts (hunk 4 above)
Comment (jmthibault79) [reply e2e/app/page/profile-page.ts:19]: Which is preferred / best practices?

The advantage of the testDataId is that we can change labels without changing tests ... but the disadvantage is that we may accidentally change labels and not catch it.

File: e2e/app/page/profile-page.ts (hunk 6 above)
Comment (jmthibault79) [reply e2e/app/page/profile-page.ts]: I want to have a generic check that works whether or not something is present - so I have changed this to something different

**Pull request merged**